pyscriptjs/tests/integration/test_01_basic.py

```diff
@@ -43,6 +43,27 @@ def test_python_exception(self):
         assert tb_lines[0] == "Traceback (most recent call last):"
         assert tb_lines[-1] == "Exception: this is an error"
 
+    # moved before `test_python_exception_in_event_handler`
+    # in order, fails if runs after
+    def test_dynamically_add_py_script_tag(self):
+        self.pyscript_run(
+            """
+            <script>
+                function addPyScriptTag() {
+                    let tag = document.createElement('py-script');
+                    tag.innerHTML = "print('hello world')";
+                    document.body.appendChild(tag);
+                }
+            </script>
+            <button onclick="addPyScriptTag()">Click me</button>
+            """
+        )
+        self.page.locator("button").click()
+        self.page.locator("py-script")  # wait until <py-script> appears
+
+        assert self.console.log.lines[0] == self.PY_COMPLETE
+        assert self.console.log.lines[-1] == "hello world"
+
```

Comment:
this is wrong: the `locator('py-script')` waits until it sees the the tag in the DOM, but it doesn't necessarily wait until it has been executed. For example, on my machine the test randomly fails because sometimes it does the `assert` when `hello world`  has not been printed yet.

The correct fix is to use `self.wait_for_console` to wait until something has been printed to the console:
```diff
diff --git a/pyscriptjs/tests/integration/test_01_basic.py b/pyscriptjs/tests/integration/test_01_basic.py
index 73f5061..70a4d79 100644
--- a/pyscriptjs/tests/integration/test_01_basic.py
+++ b/pyscriptjs/tests/integration/test_01_basic.py
@@ -59,8 +59,7 @@ class TestBasic(PyScriptTest):
             """
         )
         self.page.locator("button").click()
-        self.page.locator("py-script")  # wait until <py-script> appears
-
+        self.wait_for_console('hello world')
         assert self.console.log.lines[0] == self.PY_COMPLETE
         assert self.console.log.lines[-1] == "hello world"
```

I think that if you fix this, you don't have to move the tests before `test_python_exception_in_event_handler`: please apply the fix, move the test to its original location and check that everything works reliably

Reply:
Yes, this is a remnant of the past. In fact, such logic appears in quite a lot of tests, including some of the `py-repl` ones. They should all be fixed ideally.

I will apply the fix and revert the change if everything works correctly

Reply:
@antocuni `self.page.wait_for_selector("py-terminal")` is more reliable and fixed the issue for me while `self.wait_for_console('hello world')` didn't. I hope this removes the flakiness.

Reply:
The commit is present here: https://github.com/pyscript/pyscript/pull/1258/commits/826d85ab316217f54d084e94361d06455f07ba4f

Reply:
thanks!